src/main/java/com/github/davidcarboni/thetrain/json/Transaction.java

```diff
@@ -47,23 +57,101 @@
-                key = Keys.newSecretKey();
+                warn("transaction.enableEncryption: wrappedKey is blank, a new wrappedKey will be generated")
+                        .transactionID(id)
+                        .log();
+
+                try {
+                    key = Keys.newSecretKey();
+                } catch (Exception e) {
+                    error(e, "transaction.enableEncryption: error while attempting generate new secret key for transaction")
+                            .transactionID(id)
+                            .log();
+                    throw e;
+                }
+
+
                 salt = Random.salt();
-                wrappedKey = new KeyWrapper(password, salt).wrapSecretKey(key);
+                KeyWrapper keyWrapper = null;
+
+                try {
+                    keyWrapper = new KeyWrapper(password, salt);
+                } catch (Exception e) {
+                    error(e, "transaction.enableEncryption: error while attempting to create new KeyWrapper")
+                            .transactionID(id)
+                            .addParameter("passwordEmpty", StringUtils.isEmpty(wrappedKey))
+                            .addParameter("saltEmpty", StringUtils.isEmpty(salt))
+                            .log();
+                    throw e;
+                }
+
+                try {
+                    wrappedKey = keyWrapper.wrapSecretKey(key);
+                } catch (Exception e) {
+                    error(e, "transaction.enableEncryption: error while attempting to wrap secret key")
+                            .transactionID(id)
+                            .addParameter("keyWrapperEmpty", keyWrapper == null)
+                            .addParameter("passwordEmpty", StringUtils.isEmpty(password))
+                            .addParameter("saltEmpty", StringUtils.isEmpty(salt))
+                            .log();
+                }
             } else {
+
                 // Unwrap the existing key
-                key = new KeyWrapper(password, salt).unwrapSecretKey(wrappedKey);
+                info("transaction.enableEncryption: wrappedKey is not blank attempting to unwrap secret key")
+                        .transactionID(id)
+                        .log();
+                KeyWrapper keyWrapper = null;
+                try {
+                    keyWrapper = new KeyWrapper(password, salt);
+                } catch (Exception e) {
+                    error(e, "transaction.enableEncryption: error while attempting to create new KeyWrapper")
+                            .transactionID(id)
+                            .addParameter("passwordEmpty", StringUtils.isEmpty(wrappedKey))
+                            .addParameter("saltEmpty", StringUtils.isEmpty(salt))
+                            .log();
+                }
+
+                try {
+                    key = keyWrapper.unwrapSecretKey(wrappedKey);
+                } catch (IllegalStateException e) {
```

Comment:
catch any exception here instead of `IllegalStateException` specifically?

Reply:
Good spot